pkg/cluster/scaler/scaler.go

```diff
@@ -116,6 +120,24 @@ func (ws *workerScaler) scaleUp(ctx context.Context, count int64) *api.PluginErr
 			Capacity: to.Int64Ptr(count),
 		},
 	})
+	if err != nil && ws.testConfig.RunningUnderTest {
+		w := ws.log.WriterLevel(logrus.WarnLevel)
+		fmt.Fprintln(w, "scale failed")
+		// dump the full error
+		spew.Dump(w, err)
+		// dump the VMSS state
+		ss, err2 := ws.ssc.Get(ctx, ws.resourceGroup, *ws.ss.Name)
+		spew.Dump(w, ss, err2)
+		// try the scale again
+		err2 = ws.ssc.Update(ctx, ws.resourceGroup, *ws.ss.Name, azcompute.VirtualMachineScaleSetUpdate{
+			Sku: &azcompute.Sku{
+				Capacity: to.Int64Ptr(count),
+			},
+		})
```

Comment:
Is it possible for this to succeed so then we shouldn't fail below?

Reply:
yes, absolutely possible that it succeeds - for now I'd like to see the evidence of that, then will code up the retrier